pennylane/interfaces/torch.py

```diff
@@ -246,3 +258,289 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
         max_diff=max_diff,
     )
     return ExecuteTapes.apply(kwargs, *parameters)
+
+
+def pytreeify(cls):
+    """Pytree class"""
+    orig_fw = cls.forward
+    orig_bw = cls.backward
+    orig_apply = cls.apply
+
+    def new_apply(*inp):
+        # Inputs already flat
+        out_struct_holder = []
+        flat_out = orig_apply(out_struct_holder, *inp)
+        return pytree.tree_unflatten(flat_out, out_struct_holder[0])

```

Comment:
I'm not familiar with PyTorch. What's the difference between the `apply` method and the `forward` method? And why are both needed here?

Reply:
I added a comment in the PR. But you need to define your froward pass (execution) and you backward pass (gradients), the apply contains both information. We flatten in forward and unflatten in apply in order too bypass forward restriction.